ripple_path_find panics with IOUAmount overflow when scanning offers

## Summary

The `ripple_path_find` RPC method causes a server panic (`IOUAmount overflow`) when the pathfinder's `BookIndex.Build()` iterates ledger offers. This crashes the HTTP handler goroutine and drops the client connection. Caught by xrpl-hive `rpccompat-stateful` test suite (`ripple_path_find_no_path` test).

## Symptoms

- `ripple_path_find_no_path` test: `http post to http://...:5005: EOF` (connection dropped)
- `ripple_path_find_with_paths` test (preceding): returns 0 alternatives instead of finding valid paths (same panic, but Go's HTTP recovery returns empty response)

## Stack Trace

```
panic: IOUAmount overflow
  state.(*IOUAmountValue).normalize          (amount.go:139)
  state.NewIOUAmountValue                    (amount.go:112)
  state.NewIssuedAmountFromValue             (amount.go:314)
  state.ParseIOUAmountBinary                 (ripple_state.go:266)
  state.parseLedgerOffer                     (offer_entry.go:198)
  state.ParseLedgerOfferFromBytes            (offer_entry.go:252)
  pathfinder.(*BookIndex).Build.func1        (book_index.go:40)
  ledger.(*Ledger).ForEach.func1             (ledger.go:691)
  shamap.(*SHAMap).forEachUnsafe             (shamap.go:993)
  ...
  pathfinder.(*BookIndex).Build              (book_index.go:39)
  pathfinder.(*BookIndex).GetBooksByTakerPays
  pathfinder.(*Pathfinder).addBookLinks
  pathfinder.(*Pathfinder).FindPaths
  rpc/handlers.(*RipplePathFindMethod).Handle
```

## Root Cause

`BookIndex.Build()` calls `ForEach` over all ledger entries and parses each offer via `ParseLedgerOfferFromBytes`. When an offer has an IOU amount with a mantissa/exponent that produces `exponent > MaxExponent` after normalization, the `normalize()` function at `amount.go:139` **panics** instead of returning an error. This takes down the entire HTTP request.

## Possible Fixes

1. Add `recover()` in the `BookIndex.Build` `ForEach` callback to skip malformed offers
2. Validate mantissa/exponent bounds in `ParseIOUAmountBinary` before calling `NewIOUAmountValue`, returning an error instead of panicking
3. Change `normalize()` to return an error rather than panic when overflow is detected

Option 2 or 3 is preferred — panicking on bad data from the ledger should never crash an RPC handler.

## Relevant Files

- `internal/ledger/state/amount.go:139` — `normalize()` panic site
- `internal/ledger/state/ripple_state.go:266` — `ParseIOUAmountBinary`
- `internal/ledger/state/offer_entry.go:198` — `parseLedgerOffer`
- `internal/tx/payment/pathfinder/book_index.go:40` — `BookIndex.Build()`

## Test

xrpl-hive suite: `rpccompat-stateful`, tests: `ripple_path_find_no_path`, `ripple_path_find_with_paths`

--- a/internal/ledger/state/ripple_state.go
+++ b/internal/ledger/state/ripple_state.go
@@ -263,6 +263,24 @@ func ParseIOUAmountBinary(data []byte) (Amount, error) {
 		mantissa = -mantissa
 	}
 
+	// Validate bounds before calling NewIssuedAmountFromValue, which panics
+	// on overflow (matching rippled's Throw<std::overflow_error>). When parsing
+	// binary data from the ledger, a panic would crash the server. Return an
+	// error instead for out-of-range values that normalization cannot fix.
+	//
+	// The raw 54-bit mantissa can be up to ~1.8e16. Normalization divides by 10
+	// until mantissa <= MaxMantissa (9.999e15), incrementing exponent each time.
+	// If mantissa > MaxMantissa and exponent >= MaxExponent, normalization would
+	// need to increment exponent past MaxExponent, causing overflow. Similarly,
+	// if exponent already exceeds MaxExponent, it will always overflow.
+	absMantissa := mantissa
+	if absMantissa < 0 {
+		absMantissa = -absMantissa
+	}
+	if absMantissa != 0 && (exponent > MaxExponent || (exponent >= MaxExponent && absMantissa > MaxMantissa)) {
+		return Amount{}, fmt.Errorf("IOU amount overflow: mantissa %d exponent %d exceeds representable range", mantissa, exponent)
+	}
+
 	return NewIssuedAmountFromValue(mantissa, exponent, currency, issuer), nil
 }
 

--- a/internal/tx/payment/pathfinder/book_index.go
+++ b/internal/tx/payment/pathfinder/book_index.go
@@ -34,9 +34,19 @@ func (bi *BookIndex) Build() {
 	}
 	bi.built = true
 
-	// Walk all ledger entries looking for offers
+	// Walk all ledger entries looking for offers.
+	// The recover() safety net ensures that if any ledger entry causes a panic
+	// during parsing (e.g., IOUAmount overflow from malformed data), the entry
+	// is skipped rather than crashing the entire RPC handler goroutine.
 	seen := make(map[[2]payment.Issue]bool)
-	_ = bi.ledger.ForEach(func(key [32]byte, data []byte) bool {
+	_ = bi.ledger.ForEach(func(key [32]byte, data []byte) (cont bool) {
+		defer func() {
+			if r := recover(); r != nil {
+				// Skip this entry — malformed data should not crash the server
+				cont = true
+			}
+		}()
+
 		offer, err := state.ParseLedgerOffer(data)
 		if err != nil {
 			return true // not an offer, continue

